fdbclient/NativeAPI.actor.cpp

```diff
@@ -7978,10 +7979,8 @@ ACTOR Future<Standalone<VectorRef<BlobGranuleChunkRef>>> readBlobGranulesActor(
 				           e.name());
 			}
 			// worker is up but didn't actually have granule, or connection failed
-			if (e.code() == error_code_wrong_shard_server || e.code() == error_code_connection_failed ||
-			    e.code() == error_code_unknown_tenant) {
-				// need to re-read mapping, throw transaction_too_old so client retries. TODO better error?
-				throw transaction_too_old();
+			if (e.code() == error_code_wrong_shard_server || e.code() == error_code_connection_failed) {
+				throw blob_granule_request_failed();
```

Comment:
I have to admit I don't understand much of what is done here, but I am wondering, if connection_failed is not retriable, why blob_granule_request_failed and wrong_shard_server should be retriable? "Request failed" is also a kind of unspecific error. If we could provide more informative errors codes, it would be helpful for troubleshooting in test and prod environments.

Reply:
Thanks for explaining! If it is an internal error the client does not need to differentiate, it then probably good enough to keep this name. In any case better than transaction_too_old.